Replying to [comment:4 jrudolph]:
> This was supposedly fixed by r15927. But since the commented code was resurrected in r16328 this bug is reappearing.

See also scala/bug#2083, which to my frustration was closed wontfix.  (It also arose in the course of what I think should be safe reflection calls.)
